[chonghaoen] iP

chonghaoen opened the pull request:
### Duke Chatbot

>"To be, or not to be that is the question." - William [Shakespeare](https://en.wikipedia.org/wiki/William_Shakespeare)

Some example task you can keep track include:
- Todos
- Deadlines
- Events

Some things to **take note**:
1. Commands format go by: event -name- /from -date- /to -date
2. Date format go by: yyyy-MM-dd HHmm
3. Task list is stored in data/duke.txt by default.

Here is the `main` function:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
Current progress:
- [x] Record deadlines, todos, events
- [x] Mark them as done or undone
- [ ] Create a _GUI_ 🐶

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

chonghaoen committed:
test

chonghaoen committed:
Level-0

chonghaoen committed:
Level-1

chonghaoen committed:
Level-2

chonghaoen committed:
Level-3

chonghaoen committed:
Level-4

chonghaoen committed:
A-TextUiTesting

chonghaoen committed:
Level-5

chonghaoen committed:
Level-6

chonghaoen committed:
Add saving of task list in the hard disk.

chonghaoen committed:
Add saving of task list in the hard disk

chonghaoen committed:
Merge branch 'branch-Level-7'

chonghaoen committed:
Add parsing of date and time of tasks.

chonghaoen committed:
Add parsing of date and time of tasks

chonghaoen committed:
Add more OOP to Duke

chonghaoen committed:
Add packaging of classes

chonghaoen committed:
Add gradle support

chonghaoen committed:
Add JUnit Tests for Parser and TaskList

chonghaoen committed:
Package App as JAR file

chonghaoen committed:
Merge branch 'branch-Level-7'

chonghaoen committed:
Merge branch 'branch-Level-8'

chonghaoen committed:
Add JavaDocs comments

chonghaoen committed:
Tweak code to follow coding standard

chonghaoen committed:
Add capability to search for task with a keyword

chonghaoen committed:
Merge branch 'branch-A-JavaDoc'

chonghaoen committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/UnmarkCommand.java
#	src/main/java/duke/task/Todo.java
#	src/main/java/duke/utility/Ui.java

chonghaoen committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/command/FindCommand.java
#	src/main/java/duke/utility/Parser.java

chonghaoen committed:
Add Checkstyle support using Gradle

chonghaoen committed:
Merge branch 'branch-A-Checkstyle'

chonghaoen committed:
Add GUI to Duke

chonghaoen committed:
Merge branch 'branch-Level-10'

qinxutan commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,43 @@
+/**
+ * Represents a command to add a {@link Task} in a {@link TaskList}.
+ */
+public class AddCommand extends Command {
```
very well organised! I like how you divided the commands into different classes which made the overall code more readable :)

qinxutan replied:
very well organised! I like how you divided the commands into different classes which made the overall code more readable :)

qinxutan commented on src/main/java/duke/gui/DialogBox.java:
```diff
@@ -0,0 +1,63 @@
+ * This control represents a dialog box consisting of an ImageView to represent the speaker's face and a label
+ * containing text from the speaker.
+ */
+public class DialogBox extends HBox {
```
hahahh i like the images used!

qinxutan reviewed (commented):
LGTM! all the best for the rest of your ip :)

xuwen-ho commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    public static void main(String[] args) {
+        new Duke("data/duke.txt").run();
```
OOP-wise your code is very organised and clear, with different features split into many different classes

xuwen-ho commented on src/main/java/duke/utility/Storage.java:
```diff
@@ -0,0 +1,77 @@
+        ArrayList<Task> list = new ArrayList<>();
+        while (sc.hasNextLine()) {
+            String line = sc.nextLine();
+            if (line == "\n") {
```
nit: maybe .equals would be better than using ==

xuwen-ho commented on src/main/java/duke/utility/Parser.java:
```diff
@@ -0,0 +1,164 @@
+        }
+        String name = s.substring(5);
+        return new Todo(name);
+    }
```
createEvent, createDeadline, createTodo seem to have some duplicated code. Maybe can be combined into a single method for simplicity?

xuwen-ho reviewed (commented):
LGTM, just some minor nitpicking. Good job!

HusseinSafwan02 commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * A GUI for Duke using FXML.
+ */
+public class Main extends Application {
```
I really like how you structure your code, good work!

HusseinSafwan02 reviewed (commented):
Well done! I love your touch of personality for your duke!

chonghaoen committed:
Add assertions

chonghaoen committed:
Improve code quality

Many magic numbers in Parser when parsing commands.

Vague variable names found like tasklist referring to a file.

Let's replace magic numbers with constants and replace vague
variable names.

This will allow for more understandable code for readers.

chonghaoen committed:
Add function to archive task list

Users are currently unable to store a set of tasks for future
reference

Let's add the function to archive the current task list in a file
and clear the task list.

This will allow users to remove items and yet still have a record
somewhere.

chonghaoen committed:
Merge pull request #2 from chonghaoen/branch-A-Assertions

Add assertions

chonghaoen committed:
Merge branch 'master' into branch-A-CodeQuality

chonghaoen committed:
Merge pull request #3 from chonghaoen/branch-A-CodeQuality

Improve code quality

chonghaoen committed:
Merge branch 'branch-C-Archive'

# Conflicts:
#	src/main/java/duke/utility/Parser.java
#	src/main/java/duke/utility/Storage.java

chonghaoen committed:
Add unique personality of Chad

chonghaoen committed:
Edit README

chonghaoen committed:
Edit README

chonghaoen committed:
Edit README
